lib/Controller/DisplayGroup.php

```diff
@@ -2869,6 +2873,90 @@ public function triggerWebhook(Request $request, Response $response, $id)
             'id' => $displayGroup->displayGroupId
         ]);
 
+        return $this->render($request, $response);
+    }
+
+    /**
+     * @param Request $request
+     * @param Response $response
+     * @param null $displayGroupId
+     * @return ResponseInterface|Response
+     * @throws ControllerNotImplemented
+     * @throws GeneralException
+     * @throws NotFoundException
+     * @throws PlayerActionException
+     * @SWG\Post(
+     *   path="/displaygroup/criteria[/{displayGroupId}]",
+     *   operationId="ScheduleCriteriaUpdateActionPushCriteriaUpdate",
+     *   tags={"displayGroup"},
+     *   summary="Action: Push Criteria Update",
+     *   description="Send criteria updates to the specified DisplayGroup or to all displays if displayGroupId is not provided.",
+     *   @SWG\Parameter(
+     *       name="displayGroupId",
+     *       in="path",
+     *       description="The display group id",
+     *       type="integer",
+     *       required=false
+     *    ),
+     *   @SWG\Parameter(
+     *       name="criteriaUpdates",
+     *       in="body",
+     *       description="The criteria updates to send to the Player",
+     *       required=true,
+     *       @SWG\Schema(
+     *           type="array",
+     *           @SWG\Items(
+     *               type="object",
+     *               @SWG\Property(property="metric", type="string"),
+     *               @SWG\Property(property="value", type="string"),
+     *               @SWG\Property(property="ttl", type="integer")
+     *           )
+     *       )
+     *    ),
+     *   @SWG\Response(
+     *       response=204,
+     *       description="Successful operation"
+     *   ),
+     *   @SWG\Response(
+     *       response=400,
+     *       description="Invalid criteria format"
+     *   )
+     *  )
+     */
+    public function pushCriteriaUpdate(Request $request, Response $response, $displayGroupId = null): Response|ResponseInterface
+    {
+        $sanitizedParams = $this->getSanitizer($request->getParams());
+
+        // Get criteria updates
+        $criteriaUpdates = $sanitizedParams->getArray('criteriaUpdates');
+
+        // ensure criteria updates exists
+        if (empty($criteriaUpdates)) {
+            throw new InvalidArgumentException(__('No criteria found.'), 'criteriaUpdates');
+        }
+
+        if ($displayGroupId !== null) {
+            // fetch displays under the Display Group
+            $displayGroup = $this->displayFactory->getByDisplayGroupId($displayGroupId);
+        } else {
+            // fetch all displays regardless of Display Group
+            $displayGroup = $this->displayFactory->query();
+        }
+
+        // Create and send the player action
+        $this->playerAction->sendAction(
+            $displayGroup,
+            (new ScheduleCriteriaUpdateAction())->setCriteriaUpdates($criteriaUpdates)
```

Comment:
We should sanitize the input in the array before sending it, to make sure someone doesn't try to send unwanted parameters to the player